### ISSUE TITLE
Error in testing demo data

### ISSUE THREAD

**Issue by g0079** [agnesblch/MetIDfyR#3]
When I test MetIDfyR using demo data, R console returned some error messages and no results were generated.
When used the following command:
system("Rscript MetIDfyR.R -i input/lgd_DIA_peak-picked.tsv -o LGD4033_results -c input/config_LGD.R"),
the R console reported the following errors:
Warning messages:
1: In trimMz_Spectrum(object, mzlim = mz, msLevel. = msLevel., ...) :
  No data points between 339.089267582415 and 339.096049435585 for
spectrum with acquisition number 5661. Returning empty spectrum.
2: In trimMz_Spectrum(object, mzlim = mz, msLevel. = msLevel., ...) :
  No data points between 339.089267582415 and 339.096049435585 for
spectrum with acquisition number 5721. Returning empty spectrum.
There were 50 or more warnings (use warnings() to see the first 50)
…
Error in { : task 1 failed - "could not find function "ggarrange""
Calls: %dopar% -> <Anonymous>
Stop execution
Error in unserialize(node$con) :  error reading from connection
Calls: local ... doTryCatch -> recvData -> recvData.SOCKnode -> unserialize
also: Warning messages:
1: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
2: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
However, I'm sure that I have installed the package "ggpubr".

When used the following command:
system("Rscript MetIDfyR.R -i input/cocaine_peak-picked.tsv -o Cocaine_results -c input/config_cocaine.R")
, the following errors were reported:
Error in checkForRemoteErrors(val) : 
  one node produced an error: ERROR: failed to lock directory 'C:\Users\qingyu02\OneDrive' for modifying
Try removing 'C:\Users\qingyu02\OneDrive/00LOCK'
Calls: clusterApply -> staticClusterApply -> checkForRemoteErrors
Stop execution

When used the following command:
system("Rscript MetIDfyR.R -i input/diclofenac-picked.tsv -o Diclofenac_results -c input/config_diclofenac.R")
, the following errors were reported:
Error in checkForRemoteErrors(val) : 
  one node produced an error: ERROR: failed to lock directory 'C:\Users\qingyu02\OneDrive' for modifying
Try removing 'C:\Users\qingyu02\OneDrive/00LOCK'
Calls: clusterApply -> staticClusterApply -> checkForRemoteErrors
Stop execution

There are no results in results folder.

**Comment by agnesbrnb** [comment]
Thank you for your interest in MetIDfyR. 

I could not reproduce your errors on my machine so it may be an environment issue.
First what is your R version ? 
It appears that ggpubr is not found by pacman. What return the command "pacman::p_load("ggpubr")" ?

Also, do you get this print in R console or do you get the error/warning immediatly :

![image](https://user-images.githubusercontent.com/44233550/141086056-05d7c64c-9adc-4e47-b159-c86733870f86.png)

The error concerning Cocaine and Diclofenac could be solve as said by removing 'C:\Users\qingyu02\OneDrive/00LOCK'. 

Could you try to clone MetIDfyR in a local directory instead of OneDrive ? There may be a conflict with library location. 

Let me know if something help or if you find a solution.

**Comment by g0079** [comment]
Yes, I think you are right. Last night I removed all the project libraries, and re-installed them all in a new location, then called ggpubr and ggplot2 manually before I restarted MetIDfyR. Happily, the first task, which is "system("Rscript MetIDfyR.R -i input/lgd_DIA_peak-picked.tsv -o LGD4033_results -c input/config_LGD.R")" passed successfully. However, two new error messages occurred for the last two tasks. The error messages are the same as following:
Error in unserialize(node$con) : error reading from connection
Calls: parLapply ... FUN -> recvData -> recvData.SOCKnode -> unserialize
Execution halted
[1] 1
Thank you very much for your kindly reply.

**Comment by g0079** [comment]
Sorry for my carelessness, my R version is 4.1.2, and I will try to clone MetIDfyR in a local directory.

**Comment by g0079** [comment]
![image](https://user-images.githubusercontent.com/92711500/141221573-55e2aa98-dbe8-48ba-97dc-fe5f9fe3e5da.png)
I am sorry but the error messages still occured.

**Comment by agnesbrnb** [comment]
I manage to reproduce your issue on my local machine (4-cores 16gb). However when testing on my debian server (12-core 48gb) with only one core I didn't get this error. It seems to be a memory issue depending on the computer configuration.

After running multiple config file, cocaine with 3 transformations and diclofenac with 2 transformations are working. 

Let me now if this work for you. You could also switch "bool_phase_2" parameters to FALSE in Diclofenac config file to reduce combinations. I'll work on an optimized way to do that part of MetIDfyR.

**Comment by g0079** [comment]
I modified the Diclofenac config file as your suggestion on my office Windows laptop (Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz   1.90 GHz, 16 GB ), but the follow errors occurred: 
![image](https://user-images.githubusercontent.com/92711500/141925926-cb9d38af-519b-4277-bbc3-3e95fde8bdc6.png)

Next, I will try with a higher configuration Windows computer, because I have successfully run all three tasks on my Mac Book Pro.

**Comment by g0079** [comment]
I tried to run MetIDfyR on a Windows computer with higher configuration (Intel(R) Xeon(R) W-2133 CPU @ 3.60GHz   3.60 GHz, 64GB), but a new error occured as following:
![image](https://user-images.githubusercontent.com/92711500/142163494-35fdd581-768f-4aac-99b6-79a3e99e6677.png)
, and the error also occured for the other two tasks.

**Comment by agnesbrnb** [comment]
Problem solved. Let me know for the memory issue.

**Comment by g0079** [comment]
These are the errors on my laptop (Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz 1.90 GHz, 16 GB):
system("Rscript MetIDfyR.R -i input/cocaine_peak-picked.tsv -o Cocaine_results -c input/config_cocaine.R")
![image](https://user-images.githubusercontent.com/92711500/142335417-b8c50865-7109-40d6-8ab2-b1ab66140749.png)
system("Rscript MetIDfyR.R -i input/diclofenac-picked.tsv -o Diclofenac_results -c input/config_diclofenac.R")
![image](https://user-images.githubusercontent.com/92711500/142335490-ea0930c9-f777-42ac-875b-b6f1dcc443b6.png)
And no results were returned.

Thees are the errors on my desktop (Intel(R) Xeon(R) W-2133 CPU @ 3.60GHz 3.60 GHz, 64GB):
system("Rscript MetIDfyR.R -i input/lgd_DIA_peak-picked.tsv -o LGD4033_results -c input/config_LGD.R")
![image](https://user-images.githubusercontent.com/92711500/142335721-b12b0083-25b1-47c0-81fd-327db06850b2.png)
system("Rscript MetIDfyR.R -i input/cocaine_peak-picked.tsv -o Cocaine_results -c input/config_cocaine.R")
![image](https://user-images.githubusercontent.com/92711500/142335754-4809f14f-c647-4021-a0ef-f7cab3238b7e.png)
system("Rscript MetIDfyR.R -i input/diclofenac-picked.tsv -o Diclofenac_results -c input/config_diclofenac.R")
![image](https://user-images.githubusercontent.com/92711500/142335776-95d7e022-c9ad-44ee-9759-652e6449d7ee.png)
Only a result named out_LGD_DIA_peak-picked.tsv was returned.

**Comment by g0079** [comment]
Could you help me check if I've done something wrong when installed and used MetIDfyR on Windows 10 computer?
First, I used the following command to set the directory to intall the dependency packages in C:\Program Files\R\R-4.1.2\library:
![image](https://user-images.githubusercontent.com/92711500/142382133-84f5119d-1d6a-4e96-9fca-57bf8f6a6831.png)
Second, by using version control, I clone the MetIDfyR to C:\R\R Project\Basic\MetIDfyR. 
And then, I found the command setwd('MetIDfyR-master/') did not work because the directory is not right.
Next, I followed the instruction to unzip the data files, and copy the three command lines to R console to run MetIDfyR.
It dose not work on my Windows 10 computer, the errors were all listed above, neither laptop nor desktop, but it works well on my Mac Book Pro.

**Comment by agnesbrnb** [comment]
Your installation of MetIDfyR seems good to me. The command setwd('MetIDfyR-master/') is used when you download the zip file from Github. It's better to clone the project to get updates. 
What is the configuration of your MacBook Pro ?

It seems to be a Windows issue since I can produce your error only on my Windows desktop. I'm working on a solution.

**Comment by agnesbrnb** [comment]
Do you use R 4.1.2 on your MacBook ?

I tried different version of R on my Windows desktop (3.6.1, 4.0.5, 4.1.1 and 4.1.2). MetIDfyR ran correctly under 3.6.1 and 4.0.5. However I get the error "error reading from connection" with 4.1 versions. After testing manually the script, I conclude that there must be a conflict when multiple nodes read the same mzML file in 4.1 versions. I don't know yet if this issue come from MSnbase or R version. 

I can just tell you that on R 4.0.5 with MSnbase 2.15.7 it worked normally on my Windows computer (with adapted number of transformations).

**Comment by agnesbrnb** [comment]
The final test this morning shows that the conflict come from R 4.1. I used R 4.1.1 with MSnbase 2.15.7 (old version working on R 4.0.5) and MetIDfyR returns the same message "error reading from connection".

**Comment by g0079** [comment]
Yesterday I tested MetIDfyR on R 4.0.5 with both MSnbase 2.15.7 and 2.20.1, and changed nb_transformation to 3 (for 4, the error still occurred), it worked very well for all the three tasks. I also tested my own data, MetIDfyR can fulltill its tasks completely. The R version on my MacBook Pro (2.7 GHz dual-core Intel Core i-5, 8GB, macOS Monterey) is 4.1.2, so it seems that the combination of Windows 10 and R 4.1.2 results in the issue. Thank you very much for your kindly help.
